modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/index/DuplicateQueryStringsDetectorImpl.java

```diff
@@ -77,10 +79,18 @@ protected BooleanQuery getCriteriaQuery(Criteria criteria) {
 		_addQueryClauses(
 			booleanQuery::addFilterQueryClauses, getQueryStringsQuery(criteria),
 			getIndexQuery(criteria));
-		_addQueryClauses(
-			booleanQuery::addMustNotQueryClauses,
-			queries.term(RankingFields.INACTIVE, true),
-			getUnlessRankingIdQuery(criteria));
+
```

Comment:
i dont think we need to make this change at all.
at first i was going to suggest returning early if _resultRankingsConfiguration.allowDuplicateQueryStrings() is true (see above comment on line 55), but then i checked the usages of detect in EditRankingMVCActionCommand, and it looks like we already return early https://github.com/BryanEngler/liferay-portal/blob/29b4293a37b899069241d36a53adc09ccd3c57d5/modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/portlet/action/EditRankingMVCActionCommand.java#L287
https://github.com/BryanEngler/liferay-portal/blob/29b4293a37b899069241d36a53adc09ccd3c57d5/modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/portlet/action/EditRankingMVCActionCommand.java#L300
so the logic here would never even be hit.

Reply:
Ah I see...so if we want inactive rankings to be checked for duplicates too we'd just need to remove `queries.term(RankingFields.INACTIVE, true)?

Reply:
I'll take a look at this again, since while testing it out I was able to create duplicate rankings as long as they were inactive, but then I was able to activate duplicate rankings afterwards.

So maybe duplicate inactive rankings are fine as long as you can only activate one at a time.

Reply:
Got it. I'll drop these changes and add `guardDuplicateQueryString(actionRequest, editRankingMVCActionRequest);` to `com.liferay.portal.search.tuning.rankings.web.internal.portlet.action.EditRankingMVCActionCommand#doUpdate`.